src/metadata-builder/EntityMetadataBuilder.ts

```diff
@@ -556,30 +557,57 @@ export class EntityMetadataBuilder {
     protected createEmbeddedsRecursively(entityMetadata: EntityMetadata, embeddedArgs: EmbeddedMetadataArgs[]): EmbeddedMetadata[] {
         return embeddedArgs.map(embeddedArgs => {
             const embeddedMetadata = new EmbeddedMetadata({ entityMetadata: entityMetadata, args: embeddedArgs });
-            const targets = MetadataUtils.getInheritanceTree(embeddedMetadata.type);
+            if (typeof embeddedMetadata.type === "function") {
```

Comment:
why do we need these changes? Basically, we shouldn't change this file and only `EntitySchemaTransformer` should be changed.

Reply:
as I wrote below, we need to make embeddedMetadata.type equal to Function | undefined then we can rewrite this code just a bit and avoid duplication.